### DISCUSSION TITLE
How to center a window when scale is set to 200%

### DISCUSSION THREAD

**Discussion by technelak** [dotMorten/WinUIEx#110] [Q&A]
Hello,

I am trying to center a child window in the MainWindow of my Winui3 project. The code only works when my display scaling is set to 100%. Is there a better way to do this for it to work no matter what my display is scaled to?

When I create a new child window I get the App.MainWindow 

`var hWnd = WinRT.Interop.WindowNative.GetWindowHandle(App.MainWindow);
 var windowId = Microsoft.UI.Win32Interop.GetWindowIdFromWindow(hWnd);
 var mainAppWindow = AppWindow.GetFromWindowId(windowId);

var appWidth = mainAppWindow.Size.Width;
var appHeight= mainAppWindow.Size.Height;
var appPositionX = mainAppWindow.Position.X;
var appPositionY = mainAppWindow.Position.Y;

//Width and height of the child window
var scaledWidth = appWidth * 0.80;
var scaledHeight = appHeight * 0.80;

//child window position           
var centeredPositionX = appPositionX + ((appWidth / 2) - (scaledWidth / 2)); ;
var centeredPositionY = appPositionY + ((appHeight / 2) - (scaledHeight / 2));            
                
pdcWindow.MoveAndResize(centeredPositionX, centeredPositionY, scaledWidth, scaledHeight);

pdcWindow.BringToFront();
pdcWindow.Activate();
`

**Comment by dotMorten** [comment]
MoveAndResize already handles scale. However you could also just call `CenterOnSCreen()`

**Reply by technelak** [reply]
That's what I thought when looking at the code but not the results I was getting. After thinking about it for a few more minutes I think the issue may be how I'm setting the height and width of the child window. Awesome library by the way. I really appreciate your work!
